CefSharp.BrowserSubprocess.Core/ConcurrentObjectRegistry.h

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright © 2010-2015 The CefSharp Project. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "JavascriptCallbackWrapper.h"
+
+using namespace System::Collections::Concurrent;
+
+namespace CefSharp
+{
+    namespace Internals
+    {
+        generic <typename T>
+        public ref class ConcurrentObjectRegistry
```

Comment:
The same logic has been used in two places
https://github.com/cefsharp/CefSharp/blob/master/CefSharp.BrowserSubprocess.Core/JavascriptRootObjectWrapper.h#L29 and in the JavascriptCallbackRegistry

Without this object the
- int64 SaveMethodCallback(JavascriptAsyncMethodCallback^ callback);
- bool TryGetAndRemoveMethodCallback(int64 id, JavascriptAsyncMethodCallback^% callback);
methods have to be added to the browser wrapper which are now handled by the object repository where it is kind of hard to understand to what `_lastCallback` revers with the `_methodCallbacks` and `_callbackRegistry` members. 
Furthermore, it seemd inconsistent that there is JavascriptCallbackRegistry but for async callbacks the logic is contained directly in the class.